src/components/form/model-provider/model-provider.vue

```diff
@@ -0,0 +1,72 @@
+<script>
+    // check whether an input is a of kind toggle
+    const isToggle = target => target.getAttribute('type') === 'checkbox';
+
+    export default {
+        props: {
+            value: {
+                type: [String, Number, Array, Boolean],
+                required: true,
+            },
+        },
+        computed: {
+            model: {
+                get() {
+                    return this.value;
+                },
+                set(model) {
+                    this.$emit('input', model);
+                },
+            },
+        },
+        methods: {
+            updateValue({ target }) {
+                if (isToggle(target)) {
+                    this.updateToggle(target);
+                } else {
+                    this.model = target.value;
+                }
+            },
+            updateValueRaw(value) {
+                this.model = value;
+            },
+            updateToggle({ checked, value }) {
+                if (typeof this.value === 'boolean') {
+                    this.model = checked;
+                } else if (Array.isArray(this.model)) {
+                    if (!value) {
+                        throw new Error('I was unable to update the model. Please make sure your input has the value attribute set');
```

Comment:
I've streamlined the usage of `throw` without the `new` keyword because it's not necessary. Could you do a quick search and replace and remove it when throwing an error?

Reply:
I have done a bit of research and apparently [you](https://github.com/dreipol/dreiup-icon-checker/blob/13eebc020ee547f87bac6b5584e363f82d914d71/node/tasks/export.js#L83) also [use](https://github.com/dreipol/sib-website/blob/cd79b91239967e2ae97a4bf2edb8900c84611b33/gulp/util/errors-store.js#L29) the [`new`](https://github.com/dreipol/google-apps-script-helpers/blob/master/gulp/util/throw-plugin-error.js#L4) keyword in other contexts. I have only found [this](https://github.com/dreipol/dreiup-templates/blob/d14ca2cebcab0659c44a4950a43ff090d7aa4b69/_frontend/%7BPROJECT%7D/assets/src/js/app/util/dom/cookie.js#L59) factory method without the `new` keyword. In any case I would like to update this code only if it's strictly necessary. At moment it doesn't seem so.